src/core/internal/abort.d

```diff
@@ -0,0 +1,44 @@
+module core.internal.abort;
+
+/*
+ * Use instead of assert(0, msg), since this does not print a message for -release compiled
+ * code, and druntime is -release compiled.
+ */
+void abort(string msg, string filename = __FILE__, size_t line = __LINE__) @nogc nothrow @safe
+{
+    // use available OS system calls to print the message to stderr
+    version(Posix)
+    {
+        import core.sys.posix.unistd: write;
+        static void writeStr(const(char)[][] m...) @nogc nothrow @trusted
+        {
+            foreach(s; m)
+                write(2, s.ptr, s.length);
+        }
+    }
+    else version(Windows)
+    {
+        import core.sys.windows.windows: GetStdHandle, STD_ERROR_HANDLE, WriteFile, INVALID_HANDLE_VALUE;
+        auto h = (() @trusted => GetStdHandle(STD_ERROR_HANDLE))();
+        if(h == INVALID_HANDLE_VALUE)
+            // attempt best we can to print the message
+            assert(0, msg);
+        void writeStr(const(char)[][] m...) @nogc nothrow @trusted
+        {
+            foreach(s; m)
+            {
+                assert(s.length <= uint.max);
+                WriteFile(h, s.ptr, cast(uint)s.length, null, null);
+            }
+        }
+    }
+    else
+        static assert(0, "Unsupported OS");
+
+    import core.internal.string;
+    UnsignedStringBuf strbuff;
+
+    // write an appropriate message, then abort the program
+    writeStr("Aborting from ", filename, "(", line.unsignedToTempString(strbuff, 10), ") ", msg);
+    assert(0);
```

Comment:
Let's please use abort [¹](http://www.cplusplus.com/reference/cstdlib/abort/),[²](https://msdn.microsoft.com/en-us/library/vstudio/k089yyh0(v=vs.100).aspx) or _Exit [¹](http://www.cplusplus.com/reference/cstdlib/_Exit),[²](https://msdn.microsoft.com/en-us/library/vstudio/6wdz5232(v=vs.100).aspx) instead of the `hlt` instruction.